app/cdap/components/DataPrep/Directives/ChangeDataType/DecimalOptions.tsx

```diff
@@ -0,0 +1,293 @@
+/*
+ * Copyright © 2023 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+import React, { useState } from 'react';
+import styled from 'styled-components';
+import T from 'i18n-react';
+
+import {
+  ISubmenuProps,
+  SubmenuContainer,
+} from 'components/DataPrep/Directives/ChangeDataType/submenu';
+
+import { makeStyles } from '@material-ui/core/styles';
+import Paper from '@material-ui/core/Paper';
+import FormControl from '@material-ui/core/FormControl';
+import InputLabel from '@material-ui/core/InputLabel';
+import OutlinedInput from '@material-ui/core/OutlinedInput';
+import InputAdornment from '@material-ui/core/InputAdornment';
+import IconButton from '@material-ui/core/IconButton';
+import HelpIcon from '@material-ui/icons/Help';
+import Select from '@material-ui/core/Select';
+import MenuItem from '@material-ui/core/MenuItem';
+import FormHelperText from '@material-ui/core/FormHelperText';
+import Button from '@material-ui/core/Button';
+import Tooltip from '@material-ui/core/Tooltip';
+
+const PREFIX = 'features.DataPrep.Directives.ChangeDataType.decimalConfig';
+const ROUNDING_PREFIX = `${PREFIX}.roundingOptions`;
+
+const ID_PREFIX = 'DataPrep-Directives-ChangeDataType-decimal';
+
+const SCALE_INPUT_ID = `${ID_PREFIX}-scaleInputId`;
+const ROUNDING_INPUT_ID = `${ID_PREFIX}-roundingInputId`;
+const ROUNDING_LABEL_ID = `${ID_PREFIX}-roundingLableId`;
+
+interface IRoundingMode {
+  value: string;
+  text: string;
+  description?: string;
+}
+
+const ROUNDING_MODES: IRoundingMode[] = [
+  {
```

Comment:
nit: Do we need to validate all these hard coded  `ROUNDING_MODES` values against any backend data in order to avoid unsupported modes?

Reply:
These are all the rounding modes supported in java. Ref: https://docs.oracle.com/javase/7/docs/api/java/math/RoundingMode.html

Ideally, we should get supported rounding modes for the dataset from the backend. But in this case, I think, this hardcoded list of rounding modes should work.

Reply:
ok,fine